deploy/helm/kubecf/assets/scripts/jobs/cf-cli-6-linux/setup_internal_security_group.sh

```diff
@@ -3,60 +3,218 @@
+        --cacert "${uaa_ca_cert_pem}" \
+        --request POST \
+        --data "${scope}&${grant_type}&${client_id}&${client_secret}" \
+        "${UAA_URL}/oauth/token" \
+        | jq -r '.access_token'
+}
+
+# check_security_group checks if the security group already exists.
+function check_security_group() {
+    local access_token
+    access_token="$1"
+    local total_results
+    total_results=$(
+        curl \
+            --silent --show-error --fail \
+            --cacert "${cf_api_ca_cert_pem}" \
+            --header "Authorization: Bearer ${access_token}" \
+            "${CF_API}/v2/security_groups?q=name:${sec_group_name}" \
+            | jq -r '.total_results'
+    )
+    if [[ "${total_results}" == "0" ]]; then return 1; fi
+}
+
+# get_sec_group_id returns the security group ID.
+function get_sec_group_id() {
+    local access_token
+    access_token="$1"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${cf_api_ca_cert_pem}" \
+        --header "Authorization: Bearer ${access_token}" \
+        "${CF_API}/v2/security_groups?q=name:${sec_group_name}" \
+        | jq -r '.resources[0].metadata.guid'
+}
+
+# create_security_group creates the new security group.
+function create_security_group() {
+    local access_token
+    access_token="$1"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${cf_api_ca_cert_pem}" \
+        --header "Authorization: Bearer ${access_token}" \
+        --header "Content-Type: application/json" \
+        --request POST \
+        --data "${sec_group_json}" \
+         "${CF_API}/v2/security_groups" \
+        1> /dev/null
+}
+
+# update_security_group updates the security group with the up-to-date JSON definition.
+function update_security_group() {
+    local access_token
+    access_token="$1"
+    local sec_group_id
+    sec_group_id="$2"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${cf_api_ca_cert_pem}" \
+        --header "Authorization: Bearer ${access_token}" \
+        --header "Content-Type: application/json" \
+        --request PUT \
+        --data "${sec_group_json}" \
+        "${CF_API}/v2/security_groups/${sec_group_id}" \
+        1> /dev/null
+}
+
+# delete_security_group deletes the security group. It assumes the staging and running configs are
+# unbound.
+function delete_security_group() {
+    local access_token
+    access_token="$1"
+    local sec_group_id
+    sec_group_id="$2"
+    curl \
+        --silent --show-error --fail \
+        --cacert "${cf_api_ca_cert_pem}" \
+        --header "Authorization: Bearer ${access_token}" \
+        --request DELETE \
+        "${CF_API}/v2/security_groups/${sec_group_id}" \
+        1> /dev/null
+}
+
+# bind_security_group binds the security group to the staging and running configs.
+function bind_security_group() {
+    local access_token
+    access_token="$1"
+    local sec_group_id
+    sec_group_id="$2"
+    local configs
+    configs=(
```

Comment:
I think [https://github.com/koalaman/shellcheck/wiki/SC2155](SC2155), the declare/assign shellcheck warning, only applies to things that can fail.  There should be no issue with `local access_token="${1}"` or `local configs=(…)` (because neither actually runs commands).

Reply:
Fixed with https://github.com/cloudfoundry-incubator/kubecf/pull/485/commits/da0e3bd19f83bf58ffaa06ff6972d0ed46930e8e.